### ISSUE TITLE
openstudio::SqlFile::energyConsumptionByMonth does not work for fuel type of "Additional Fuel"

### ISSUE THREAD

**Issue by DavidGoldwasser** [NREL/OpenStudio#3381]
There was a recent bug relate to fuel oil 1 being used in calibration measure. That has been fixed, but I stumbled across issue getting monthly results by fuel end use and category in OpenStudio Results. Fuel oil is lumped in with "Additional Fuel". 

To demonstrate this I added fuel oil to a test model and added two asserts in the test. One assert is that it is finding fuel oil on a `sqlFile.execAndReturnFirstDouble ` of tabular annual data for "Additional Fuel". This assert passes. 

I then added a second assert using `sqlFile.energyConsumptionByMonth` for "Additional Fuel", 0 for category, and 1 for month. This are not initialized and do not report any non-zero values for any end uses or months.
https://github.com/NREL/OpenStudio-measures/commit/c38197a993fb04b8713c9486446c96e55e6234c8#diff-0f437873348effabf1a199c8b6587c0fR345

**Comment by DavidGoldwasser** [comment]
The EnergyPlus HTMl file only has "BUILDING ENERGY PERFORMANCE" table for electricity and natural gas, so if OpenStudio is leaning on this, then that can explain why additional fuel doesn't work.

If that is the case, then maybe we can request that table in EnergyPlus add that table for all fuel types used in the model.

**Comment by DavidGoldwasser** [comment]
If I had to put in a work around for now, it would be to add meter with monthly time step for additional fuel, or directly for fuel oil.

**Comment by DavidGoldwasser** [comment]
It does look like `energyConsumptionByMonth` does just roll up a `execAndReturnFirstDouble` query into tabular data. @macumber do you see an easy solution other than having EnergyPlus team add this table. 

https://github.com/NREL/OpenStudio/blob/2da860da42395b39845533d74c372644b1bb18c7/openstudiocore/src/utilities/sql/SqlFile_Impl.cpp#L863-L881

**Comment by DavidGoldwasser** [comment]
@asparke2 had idea to add additional `Output:Table:Monthly` I have tested this on IDF file and it does add additional tables, for meters, but I'm having problem adding it properly in energyPlusOutputRequests, I'm probably just making typo. The code below is for fuelOiL, but I would probably use additionalFuel instead,

```
    result << OpenStudio::IdfObject.load('Output:Meter,FuelOil#1:Facility,hourly;').get
    result << OpenStudio::IdfObject.load('Output:Table:MonthlyReports,2,FuelOil#1:Facility,SumOrAverage;').get
```

**Comment by asparke2** [comment]
check out the `eplusout.mdd` file from a model with fuel oil to get the right syntax for the Meter

**Comment by DavidGoldwasser** [comment]
@asparke2 sorry, the meter output request goes in fine, I used the mdd to get that, although I think I want to switch to AdditionalFuels, which is also in the mdd, as that the fuel type OpenStudio is expecting in this use. It is the MonthlyReport that isn't making it in. @macumber this is documented as a supported object type for energyPlusOutputRequests but I have not found an example if it used anywhere yet, and I couldn't figure out a syntax that works.

**Comment by DavidGoldwasser** [comment]
`I've been trying to add Output:Table:Monthly` now, not `Output:TableMonthlyReports`. This is in line with what is added to Gas and Electricity. My code looks like this.

```
    result << OpenStudio::IdfObject.load('Output:Meter,FuelOil#1:Facility,timestep;').get

    monthly_array = ['Output:Table:Monthly']
    monthly_array << 'Building Energy Performance - Fuel Oil #1'
    monthly_array << '2'
    monthly_array << 'Heating:FuelOil#1'
    monthly_array << 'SumOrAverage'
    monthly_array << 'WaterSystems:FuelOil#1'
    monthly_array << 'SumOrAverage;'

    monthly_string = monthly_array.join(",").to_s
    puts "string is"
    puts monthly_string
    asdf = OpenStudio::IdfObject.load(monthly_string).get
    puts "asdf"
    puts asdf
    result << asdf
```
I've been having issues testing this. The reporting measure test currenty reverse translates the output request, and the addes them to the OSM. `Output:Table:Monthly` is getting lost in the process. 

Here is come code from the test, confirming that while 5 objects are added to workspace, only 4 are added to model. This is from `setup_test` method.
```
    # convert output requests to OSM for testing, OS App and PAT will add these to the E+ Idf
    workspace = OpenStudio::Workspace.new('Draft'.to_StrictnessLevel, 'EnergyPlus'.to_IddFileType)
    puts "hello2 workspace starts with #{workspace.objects.size}"
    workspace.addObjects(idf_output_requests)
    puts "hello3 workspace finishes with #{workspace.objects.size}"
    rt = OpenStudio::EnergyPlus::ReverseTranslator.new
    request_model = rt.translateWorkspace(workspace)

    translator = OpenStudio::OSVersion::VersionTranslator.new
    model = translator.loadModel(model_in_path)
    assert(!model.empty?)
    model = model.get
    puts "hello3 model starts with #{model.objects.size}"
    model.addObjects(request_model.objects)
    puts "hello4 model finishes with #{model.objects.size}"
    model.save(model_out_path(test_name), true)
```

No reverse translator for object, this is also true for some other objects that can be added as output requests.
https://github.com/NREL/OpenStudio/blob/8dbcb41588a5873559b9cb06058c37c0afa33e87/openstudiocore/src/energyplus/ReverseTranslator.cpp#L740

@macumber this might just affect measure test, and not use in OS app and PAT? I'll do some testing. Once I can get this in an IDF file, I should then be able able to get new table in SQL tabular data, and runner.registerValue for FuelOil#1 and FuelOil#2.

**Comment by DavidGoldwasser** [comment]
@macumber I have this working now in current version of OpenStudio. I'll let you decide if you want to keep this open to add functionality directly to OpenStudio or close it and leave functionality as is. Below are three possible enhancements to support additional fuel.
https://github.com/NREL/OpenStudio-measures/commit/5cc163577c6e501857f1345243166eb78c7e6b33

1. While OpenStudio automatically adds Output:Table:Monthly objects for typical fuel types, it doesn't for additional fuel types such as FuelOil#1. I added them for consumption only (not demand) for all additional fuel types. I did this using the energyPlusOutputRequests section. OpenStudio could do this automatically?
2. OpenStudio doesn't support reverse translation of Output:Table:Monthy objects, or a number of other objects that are supported in energyPlusOutputRequests method. Reverse translation could be added or the testing framework in reporting measures could be updated to now rely on forward translation of these objects in test setup method. I'm not exactly what the fix in measure test would look like.
3. I updated the monthly summary consumption table fin os_lib_reporting.rb to loop through an extended_fuel_type_names list instead of the typical list. When an extended fuel type was found the SQL query was manually performed on tabular data using execAndReturnFirstDouble. energyConsumptionByMonth could be updated to accept additional fuel types, or a new energyConsumptionByMonthAdditionalFuelTypes method could be added. Or we do nothing, and just rely on the manual query I used. This only make sense to do if item #1 is also implemented for additional fuel types.

I'm fine if you choose to do all of these or none of these, however if item 2 isn't implemented then I will have to come up with a solution for supporting testing of this functionality in the measure testing framework.
